pkg/manifests/errmapreader.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manifests
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// errMapReader wraps a map[string]string and exposes read methods
+// for values and slices.
+// It tracks the first occured error in errMapReader.err
+// when reading values from the underlying map.
+type errMapReader struct {
+	src map[string]string
+	err error
+}
+
+func NewErrMapReader(src map[string]string) *errMapReader {
```

Comment:
This should not be an exported func, plus, it returns an unamable type to callers from other packages.

Reply:
agreed :+1: